### DISCUSSION TITLE
Inference time of each layer

### DISCUSSION THREAD

**Discussion by ghost** [ultralytics/yolov5#8444] [Q&A] [answered]
Hello I'm trying to profile inference time of each layer of yolov5.
I have added the code below to `yolo.py` between **Create model** and **Options**:

```
......
# Create model
im = torch.rand(opt.batch_size, 3, 640, 640).to(device)
model = Model(opt.cfg).to(device)

x = torch.randn(1, 3, 640, 640).cuda()

with torch.autograd.profiler.profile(enabled=True, use_cuda=True, record_shapes=False,
                                     profile_memory=False) as prof:
    output = model(x)
print(prof.table())
prof.export_chrome_trace('./model.json')

summary(model, x)

# Options
if opt.line_profile:   # profile layer by layer
    _ = model(im, profile=True)
......
```

the result is as follow:

![yolo-v5s](https://user-images.githubusercontent.com/29224123/177006077-71c3e491-7bdf-4283-89a3-f99c1ac2c6f7.png)

We can clearly see that the first convolutional layer takes more than 90% of the inference time, but when I search the reason from  source code, I haven't find any difference between the first conv layer and the other conv layers.

From my experiment，other models like mask-rcnn，mobileNet，vgg，the inference time of their conv layer is not much different.

### Question
So I want to know the reason of this phenomenon. Appreciate for any help!

**Comment by glenn-jocher** [comment, marked as answer]
@769430755 your profiler is incorrect. 

We already have built-in line profiling of YOLOv5 models, they display even resource utilization throughout the model generally speaking.
```bash
python models/yolo.py --cfg yolov5s.yaml --line-profile
```

<img width="1078" alt="Screen Shot 2022-07-02 at 6 42 57 PM" src="https://user-images.githubusercontent.com/26833433/177009134-8a6ea73e-7788-4136-8063-1e44af19604c.png">
